libdistopia/libdistopia/lib/distance_array_dispatch.h

```diff
@@ -0,0 +1,33 @@
+#ifndef DISTOPIA_DISTANCE_ARRAY_DISPATCH_H
+#define DISTOPIA_DISTANCE_ARRAY_DISPATCH_H
+
+#include <cstddef>
+#include <cstdio>
+#include "distopia_type_traits.h"
+
+template <typename T>
+void DistanceArrayOrthoDispatch(const T *coords0, const T *coords1, const T *box,
+                                std::size_t n0, std::size_t n1, T *out);
+
+using DistanceArrayOrtho_FptrT = decltype(&DistanceArrayOrthoDispatch<float>);
+using DistanceArrayOrtho_DptrT = decltype(&DistanceArrayOrthoDispatch<double>);
+
+enum selectT {Float = 0, Double = 1};
```

Comment:
I started hacking this together, right now it'll need to be anonymously namespaced right?  Should all this gubbins be placed somewhere central?  Or worse still, should `calc_bonds_dispatch.h` actually be `dispatch.h` and handle everything?

Reply:
I think your right that sharing over files is going to get a bit painful, particularly as `_DISPATCH_fptr_register` needs to know about every function and  be initialised in one place only. I say we go with the ugly but effective combining everything in `dispatch.h` and `dispatch.cpp` That way all of the annoying declarations of namespaces and macros live in the same header files etc.